roles/network_plugin/calico/tasks/install.yml

```diff
@@ -110,10 +110,40 @@
     - calico_pool_cidr_ipv6 is defined
     - enable_dual_stack_networks
 
+- name: Calico | Get the list of CRDs manifests
+  find:
+    paths: "{{ local_release_dir }}/calico-{{ calico_version }}-kdd-crds"
+    patterns: "^crd\\.projectcalico\\.org_.*?\\.ya?ml$"
+    use_regex: yes
+  register: calico_kdd_crds_manifests
+  when:
+    - inventory_hostname in groups['kube-master']
+    - calico_datastore == 'kdd'
+  no_log: true
+
+- name: Calico | Replace unwanted annotations and creationTimestamp keys from metadata in Calico manifests
+  lineinfile:
+    path: "{{ item.0.path }}"
+    state: absent
+    regexp: "{{ item.1 }}"
+  loop: "{{ calico_kdd_crds_manifests.files|product(lines_to_remove)|list }}"
+  loop_control:
+    label: "{{ item.0.path|basename }}"
+  vars:
+    lines_to_remove:
+      - "^\\s{2,4}annotations:$"
+      - "^\\s{2,4}controller-gen\\.kubebuilder\\.io/version: \\(devel\\)$"
+      - "^\\s{2,4}creationTimestamp: null$"
+  when:
+    - inventory_hostname in groups['kube-master']
+    - calico_datastore == "kdd"
+
 - name: Calico | Create calico manifests for kdd
-  template:
-    src: "{{ item.file }}.j2"
+  assemble:
```

Comment:
You could do the assemble first with `creates:` and the calico version in the name, and then run lineinfile, so only 4 items total (1 assemble + 3 lineinfile)

Reply:
Maybe I'm having it wrong but the `creates` keyword is not supported by all the modules and it seems that it is not supported by assemble is it ?
If creates is not usbale, having the lineinfile performed after assemble will make the "Create calico manifests for kdd" task systematically changed

Reply:
As an alternative to avoid loops and as the removal is temporary until calico 3.17, using the shell module (with sed and wildcard) would be acceptable. What do you think ?

Reply:
I changed with sed in shell module as mentioned in my last comment.